internal/dag/gatewayapi_processor.go

```diff
@@ -246,15 +247,53 @@ func (p *GatewayAPIProcessor) computeHTTPRoute(route *gatewayapi_v1alpha1.HTTPRo
 				routeAccessor.AddCondition(status.ConditionResolvedRefs, metav1.ConditionFalse, status.ReasonDegraded, fmt.Sprintf("Service %q does not exist in namespace %q", meta.Name, meta.Namespace))
 				continue
 			}
-			services = append(services, service)
+
+			var headerPolicy *HeadersPolicy
+			for _, filter := range forward.Filters {
+				switch filter.Type {
+				case gatewayapi_v1alpha1.HTTPRouteFilterRequestHeaderModifier:
+					if filter.RequestHeaderModifier.Add != nil {
+						routeAccessor.AddCondition(status.ConditionNotImplemented, metav1.ConditionTrue, status.ReasonHTTPRouteFilterType,
```

Comment:
Sorry I missed this on the first pass. We should be able to handle "add". Internally it's just a "set", if the header key doesn't exist then Envoy should add the header.

Reply:
Umm... As per gateway-api spec, `add` appends the header even with the same key name.

ref: https://gateway-api.sigs.k8s.io/spec/#networking.x-k8s.io/v1alpha1.HTTPRequestHeaderFilter

```
Input: GET /foo HTTP/1.1 my-header: foo
Config: add: {“my-header”: “bar”}
Output: GET /foo HTTP/1.1 my-header: foo my-header: bar
```

So we can handle the `add` operation when the header key does not exist, but we cannot handle it when the header key already exist as current `HeadersPolicy.Set` is `map[string]string`. I think the support of `add` operation can include the refactoring of headersPolicy.

Reply:
If the header already exists in the request Envoy will handle the adding bits. We have a setting that handles that: https://github.com/projectcontour/contour/blob/46d5ba9df6f6634143a534f8d0edd0d497cad2dc/internal/envoy/v3/route.go#L247

Reply:
I see! Updated. Than you!